allennlp/data/dataset.py

```diff
@@ -144,7 +148,17 @@ def as_arrays(self,
         # Finally, we combine the arrays that we got for each instance into one big array (or set
         # of arrays) per field.
         for field_name, field_array_list in field_arrays.items():
-            if isinstance(field_array_list[0], (list, tuple)):
+            if isinstance(field_array_list[0], dict):
+                # This is creating a dict of {namespace: batch_array} for each
+                # namespace within this field. This is mostly utilised by TextFields,
+                # which will have one namespace per TokenIndexer used within the Field.
+                namespace_batch_dict = defaultdict(list)
+                for namespace_dict in field_array_list:
+                    for namespace, array in namespace_dict.items():
+                        namespace_batch_dict[namespace].append(array)
+                field_arrays[field_name] = {namespace: numpy.asarray(array_list) for
+                                            namespace, array_list in namespace_batch_dict.items()}
+            elif isinstance(field_array_list[0], (list, tuple)):
```

Comment:
Yeah I left this as it's still possible that `ListField` would yield lists, but really I don't think there's any reason to keep this. I'll change ListField.